frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +538,189 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
   wr.code(fn, {formal});
 }
 
+/* parser errors */
+
+void ErrorParsing::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<std::string>(info);
+  assert(errorMessage.back() == '.' &&
+         "expected a period at the end of ErrorParsing message");
+  wr.heading(kind_, type_, loc, errorMessage);
+}
+
+void ErrorCannotAttachPragmas::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto stmt = std::get<const uast::AstNode*>(info);
+  wr.heading(kind_, type_, loc, "cannot attach 'pragma's to this statement.");
+  if (stmt) {
+    wr.code(stmt);
+  }
+  wr.message(
+      "Only declarations, such as variable declarations, can have 'pragma's "
+      "attached to them.");
+}
+
+void ErrorInvalidIndexExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc,
+             "only identifiers or tuples are allowed as index expressions.");
+  wr.message("Invalid index expression used here:");
+  wr.code(loc);
+}
+
+void ErrorRecordInheritanceNotSupported::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto recordName = std::get<std::string>(info);
+  wr.heading(kind_, type_, loc,
+             "inheritance is not currently supported for records.");
+  wr.note(loc, recordName, " declared as a record here");
```

Comment:
perchance a `:` at the end here? It'll get stripped in brief mode anyway.

Reply:
Good point. I'll add one for the "unterminated comment here" and "nested comment here" notes as well.